src/fifelse.c

```diff
@@ -358,12 +362,13 @@ SEXP fcaseR(SEXP na, SEXP rho, SEXP args) {
       }
     } break;
     case STRSXP: {
-      const SEXP *restrict pouts = STRING_PTR_RO(outs);
-      const SEXP pna = nonna ? STRING_PTR_RO(na)[0] : NA_STRING;
+      const SEXP *restrict pthens;
+      if (!naout) pthens = STRING_PTR_RO(thens); // the content is not useful if out is NA_LOGICAL scalar
+      const SEXP pna = NA_STRING;
       for (int64_t j=0; j<len2; ++j) {
         const int64_t idx = imask ? j : p[j];
-        if (pcons[idx]==1) {
-          SET_STRING_ELT(ans, idx, pouts[idx & amask]);
+        if (pwhens[idx & whenMask]==1) {
+          SET_STRING_ELT(ans, idx, naout ? pna : pthens[idx & thenMask]);
         } else {
           if (imask) {
             SET_STRING_ELT(ans, idx, pna);
```

Comment:
I know that for the imask case we have that `idx = j` but I would prefer we either do `pans[j] = pna` in the `if (imask)` of the `INTEGER`, `REAL`, `COMPLEX` case or `pans[idx] = pna` (just in string syntax) for the `STRING` and `VECTOR` case for consistency

Reply:
makes sense. the comments in the `VECSXP` branch tell me we shouldn't expect parity in that branch b/c of how "missing" for list columns is a bit different, so I left that be.